cibuildwheel/util.py

```diff
@@ -58,11 +59,11 @@ def __getattr__(self, attr):  # type: ignore
         return getattr(self.stream, attr)
 
 
-def download(url: str, dest: str) -> None:
-    print('+ Download ' + url + ' to ' + dest)
-    dest_dir = os.path.dirname(dest)
-    if not os.path.exists(dest_dir):
-        os.makedirs(dest_dir)
+def download(url: str, dest: Path) -> None:
```

Comment:
maybe:
```
def download(url: str, dest: Union[Path, str]) -> None:
    dest = Path(dest)
```
to not need call `Path` constructor on call of function?
```

Reply:
Why not? I think it's again good to have extra type checks and to now have `str` and `Path` be different types, or do you not agree?

Reply:
I do not know why but call `download(url, Path("/some/path"))` looks strange for me. And hide `Path` conversion still give runtime check.

But it is only my personal feel that such call is not pythonic.

Reply:
Maybe, yes. But I still don't really mind. It makes the user on the calling side think if it is really a `str` or a `Path`. But if you want, I can change it.

Reply:
I'd rather stick to Path, as we might as well be explicit about it when we can